### ISSUE TITLE
Apply more functional concepts in this library

### ISSUE THREAD

**Issue by carlosrberto** [ContaAzul/creditcard.js#79]
As you suggest @fernahh  I'm replicating my coments from https://github.com/ContaAzul/creditcard.js/pull/78 here.

I think https://github.com/ContaAzul/creditcard.js/pull/78 is a good start point for turning this library more functional.

As an improvement, but not really required I also suggest applying more funcional concepts like decomposing some functions in smaller ones with less responsibilities.

**For example `isSecurityCodeValid`:**

```javascript
export function isSecurityCodeValid(number, code) {
  let brand = getCreditCardNameByNumber(number);
  let numberLength;

  numberLength = (brand === 'Amex') ? 4 : 3;
  let regex = new RegExp(`^[0-9]{${numberLength}}$`);

  return regex.test(code);
}
```

could be refactored to something like:

```javascript

const isSecurityCodeValid = (number, code) =>
  isCodeLengthValid(code, getBrandCodeLength(getBrand(number)));
```

or using some simple functional helpers like `compose` and `curry` for better readability: 

```javascript
const isSecurityCodeValid = (number, code) =>
  compose(curry(isCodeLengthValid)(code), getBrandCodeLength, getBrand)(number);
```

**Another example `isExpirationDateValid`:**

```javascript
export function isExpirationDateValid(paramMonth, paramYearn) {
  const month = parseInt(paramMonth, 10);
  const year = parseInt(paramYearn, 10);

  const currentYear = new Date().getFullYear();
  const currentMonth = new Date().getMonth() + 1;

  if (isNaN(month) || isNaN(year))
    return false;

  if (year === currentYear && month < currentMonth)
    return false;

  if (month < 1 || month > 12)
    return false;

  return !(year < 1000 || year >= 3000);
}
```

could be:

```javascript
export const isExpirationDateValid = (month, year) => 
  isValidMonth(month) && isValidYear(year) && !isPastDate(new Date(year, month));
```

**Comment by fernahh** [comment]
@carlosrberto can you get this issue for V3?

**Comment by carlosrberto** [comment]
Yes, I can. I'll start working on this issue soon @fernahh!

**Comment by dedicio** [comment]
HI @carlosrberto 

Thanks for your contribution!

That suggestions was implemented in PR #120 

See more details: https://contaazul.github.io/creditcard.js/
